stp/constraints.lisp

```diff
@@ -141,161 +137,184 @@
+                    (eq!        (next EXO_INST)    EVM_INST_LT)
+                    (eq!        (next RES_LO)      OOGX)
+                    (eq!        (next WCP_FLAG)    1)))
 
 ;; rows of CREATE instructions that don't produce an OOGX
-(defconstraint CREATE-type-no-exception (:guard (first-row-of-unexceptional-CREATE))
-  (begin  ;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
-         ;;   ------------->   row i + 2   ;;
-         ;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
-         (vanishes! (shift ARG_1_HI 2))
-         (eq! (shift ARG_1_LO 2) (create-gDiff))
-         (eq! (shift ARG_2_LO 2) 64)
-         (eq! (shift EXO_INST 2) EVM_INST_DIV)
-         (eq! (shift MOD_FLAG 2) 1)))
-
-;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
+(defconstraint    CREATE-type---unexceptional---row-i-plus-2
+                  (:guard (first-row-of-unexceptional-CREATE))
+                  (begin  
+                    ;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
+                    ;;   ------------->   row i + 2   ;;
+                    ;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
+                    (vanishes!  (shift ARG_1_HI 2))
+                    (eq!        (shift ARG_1_LO 2)    (create-gDiff))
+                    (eq!        (shift ARG_2_LO 2)    64)
+                    (eq!        (shift EXO_INST 2)    EVM_INST_DIV)
+                    (eq!        (shift MOD_FLAG 2)    1)))
+
+;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
 ;;   Setting GAS_UPFRONT and GAS_STPD   ;;
-;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
+;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
+
 (defconstraint CREATE-type-outputs (:guard (first-row-of-CREATE))
-  (begin (eq! GAS_UPFRONT (create-gPrelim))
-         (vanishes! GAS_STIPEND)
-         (if-zero OOGX
-                  (eq! GAS_OUT_OF_POCKET
-                       (- (shift ARG_1_LO 2) (shift RES_LO 2)))
-                  (vanishes! GAS_OUT_OF_POCKET))))
+               (begin (eq!       GAS_UPFRONT (create-gPrelim))
+                      (vanishes! GAS_STIPEND)
+                      (if-zero   OOGX
+                                 (eq!       GAS_OUT_OF_POCKET    (- (shift ARG_1_LO 2) (shift RES_LO 2)))
+                                 (vanishes! GAS_OUT_OF_POCKET))))
 
 ;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
 ;;                                                ;;
 ;;    4 Constraints for CALL-type instructions    ;;
 ;;                                                ;;
 ;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
-;; all of these definitions implicitly assume that the current row (i) is such that
-;; - STAMP[i] !eq! STAMP[i - 1]       <eq! first row of the instruction
-;; - INST_TYPE eq! 0                  <eq! CALL-type instruction
-(defun (first-row-of-CALL)
-  (* (remained-constant! STAMP) (is_call)))
 
-(defun (first-row-of-unexceptional-CALL)
-  (* (first-row-of-CALL) (- 1 OOGX)))
+(defun (call---first-row-common)                 (*    (remained-constant! STAMP) (is_call)))
```

Comment:
fixes from the spec issue [STP fixes #535](https://github.com/Consensys/linea-specification-internal/issues/535)